js/controller.js

```diff
@@ -830,12 +830,16 @@ OO.plugin('Html5Skin', function(OO, _, $, W) {
 
     onVcPlay: function(event, source) {
       this.state.currentVideoId = source;
-      if (this.state.adWasPaused && this.state.currentAdsInfo && this.state.currentAdsInfo.currentAdItem && this.state.currentAdsInfo.currentAdItem.ssai) {
-        this.state.adPauseDuration = Date.now() - this.state.adPausedTime;
-        //we calculate new ad end time, based on the time that the ad was paused.
-        this.state.adEndTime = this.state.adEndTime + this.state.adPauseDuration; //milliseconds
-        this.state.adWasPaused = false;
-        this.state.adPauseDuration = 0;
+      if (this.state.currentAdsInfo.currentAdItem.ssai) {
+        //When pause come from ad source, for SSAI we still want source to be main, for consistency.
+        this.state.currentVideoId = OO.VIDEO.MAIN;
```

Comment:
Is something changing the currentVideoId from OO.VIDEO.MAIN for ssai?

Reply:
Well I realized `onVcPlay` method get called twice, one with `main` as source and another one with `ads` source.  So we are preventing `currentVideoId` is overwritten with `ads` for ssai ads.